tests/publiccloud/register_system.pm

```diff
@@ -34,14 +34,14 @@ sub run {
         # note: ssh_script_retry dies on failure
         $args->{my_instance}->retry_ssh_command("sudo SUSEConnect -r " . get_required_var('SCC_REGCODE'), timeout => 420, retry => 3);
 
+        # We need to pass the architecture of the target system, which is not set in /etc/os-release
+        my $arch = $args->{my_instance}->run_ssh_command(cmd => 'echo ${CPU}');
```

Comment:
```suggestion
        my $arch = check_var('PUBLIC_CLOUD_ARCH', 'arm64') ? 'aarch64' : 'x86_64' ; 
```
I meant something like this .

Reply:
Why not get the architecture directly from the instance instead of relying on the external setting `PUBLIC_CLOUD_ARCH`?

I find the `$CPU` environmental variable more robust here, because it is build-in and already used in `ssh_add_suseconnect_product`.

Reply:
ok